src/value.cc

```diff
@@ -511,27 +487,48 @@ std::string Value::chrString() const
   return boost::apply_visitor(chr_visitor(), this->value);
 }
 
-const Value::VectorType &Value::toVector() const
+void Value::VectorPtr::flatten() {
+  int n = 0;
+  for (unsigned int i = 0; i < (*this)->size(); i++) {
```

Comment:
Could you just do `for (const auto v : **this)` here?
Same below.

Reply:
Changed in https://github.com/openscad/openscad/pull/3096/commits/d42714a6cad756645f1ab635316fedf66f05af0d - using `std::accumulate()`

Reply:
flatten was rewritten again in https://github.com/openscad/openscad/pull/3102/commits/905884b916be5865cc1f7f8ca01262bfefbf7a03 for reason related to "embedded vectors".  
flatten is now basically one simple range-based for loop, utilizing a special iterator.

For reference, the discussion about embedded vectors starts around [my last paragraph here](https://github.com/openscad/openscad/pull/3102#issuecomment-652579183) while discussing how or where it would be safe to avoid copies of elements.